File: server/schemas/facts.py
"""
Project N: Pydantic v2 Schemas for Child Profile Facts and Therapist Techniques.
"""

from pydantic import BaseModel, Field


class FactCreateRequest(BaseModel):
    category: str = Field(
        ...,
        pattern="^(comfort_object|calming_cue|sensory_trigger|therapist_technique|communication_routine)$",
    )
    fact_title: str = Field(..., min_length=2, max_length=120)
    description: str = Field(..., min_length=5, max_length=1000)
    source_type: str = Field(..., pattern="^(home_observation|ot_session|slp_session|school)$")
    clinician_role: str | None = None
    clinician_id: str | None = None
    provenance_episode_id: str | None = None


class FactConfirmRequest(BaseModel):
    confirmed: bool = True
    custom_notes: str | None = None


class FactResponse(BaseModel):
    id: str
    category: str
    fact_title: str
    description: str
    source_type: str
    clinician_role: str | None = None
    clinician_id: str | None = None
    provenance_episode_id: str | None = None
    confirmed_by_caregiver: int
    times_tried: int
    times_helpful: int
    created_at: str | None = None


class FactListResponse(BaseModel):
    facts: list[FactResponse]
    total: int

File: ui/src/types/facts.ts
/**
 * Project N: Personal Profile Facts and Clinical Knowledge Types.
 * Strict parity with storage/db_schema.py and storage/facts_repo.py.
 */

export type FactCategory =
  | "comfort_object"
  | "calming_cue"
  | "sensory_trigger"
  | "therapist_technique"
  | "communication_routine";

export type FactSourceType =
  | "home_observation"
  | "ot_session"
  | "slp_session"
  | "school";

export interface ChildProfileFact {
  id: string;
  category: FactCategory;
  fact_title: string;
  description: string;
  source_type: FactSourceType;
  clinician_role?: string | null;
  clinician_id?: string | null;
  provenance_episode_id?: string | null;
  confirmed_by_caregiver: number; // 0 = unconfirmed, 1 = confirmed
  times_tried: number;
  times_helpful: number;
  created_at?: string;
  updated_at?: string;
}

export interface FactCreateRequest {
  category: FactCategory;
  fact_title: string;
  description: string;
  source_type: FactSourceType;
  clinician_role?: string;
  clinician_id?: string;
  provenance_episode_id?: string;
}

export interface FactConfirmRequest {
  confirmed: boolean;
  custom_notes?: string;
}
